osquery/tables/system/python_packages.cpp

```diff
@@ -130,13 +140,59 @@ void genWinPythonPackages(const std::string& keyGlob,
       if (p.at("name") != "(Default)") {
         continue;
       }
-      genSiteDirectories(p.at("data"), results, logger);
+      genSiteDirectories(p.at("data"), results, logger, user_id);
     }
     pythonInstallLocation.clear();
   }
 #endif
 }
 
+std::vector<std::map<std::string, std::variant<std::int64_t, std::string>>>
+getUserPathList(const QueryContext& context) {
+  std::vector<std::map<std::string, std::variant<std::int64_t, std::string>>>
+      paths_list;
+
+  auto user_list = usersFromContext(context);
+  for (const auto& user : user_list) {
+    if (user.count("uid") == 0 || user.count("directory") == 0) {
+      continue;
+    }
+
+    const auto& uid_as_string = user.at("uid");
+    auto uid_as_big_int = tryTo<int64_t>(uid_as_string, 10);
+    if (uid_as_big_int.isError()) {
+      LOG(ERROR) << "Invalid uid field returned: " << uid_as_string;
+      continue;
+    }
+    const auto& path = user.at("directory");
+
+    for (const auto& path_postfix : kUserDirectoryPaths) {
+      auto dir = path + "/" + path_postfix + "/";
+
+      std::vector<std::string> versions;
+      if (!listDirectoriesInDirectory(dir, versions, false).ok()) {
+        continue;
+      }
+
+      for (const auto& version : versions) {
+        auto path = version + "/lib" + "/python%" + "/site-packages";
```

Comment:
Nit:
```suggestion
        auto path = version + "/lib/python%/site-packages";
```

Reply:
I'm not sure how easily this can be refactored, but the logic around 
```
      for (const auto& version : versions) {
        auto path = version + "/lib" + "/python%" + "/site-packages";
```

Appears repeated betweean the site python packages and the user ones. So I wonder if there's some cleanup that might really help understandability

Reply:
👍 Looking at this closer I think the original code is actually broken
```
auto complete = version + "lib/python" +
                        version_path.filename().string() + "/site-packages";
```
This ends up outputting 
`/Library/Developer/CommandLineTools/Library/Frameworks/Python3.framework/Versions/3.9lib/pythonVersions/site-packages`
or 
`/Library/Developer/CommandLineTools/Library/Frameworks/Python3.framework/Versions/Currentlib/pythonVersions/site-packages`

on my mac

Reply:
Now that I have refactored that logic with the new `traverseVersions` method the paths look like this:
`/Library/Developer/CommandLineTools/Library/Frameworks/Python3.framework/Versions/3.9/lib/python3.9/site-packages/`
`/Users/ksykulev/.pyenv/versions/3.13.1/lib/python3.13/site-packages/`
`/Users/ksykulev/Library/Python/3.13/lib/python/site-packages/`

This actually gives me more results than the previous version of the code.